src/Internal/Payment/State/InitialState.php

```diff
@@ -109,6 +121,13 @@ public function start_processing( PaymentRequest $request ) {
 		// Start multiple verification checks.
 		$this->process_order_phone_number();
 
+		if ( $this->failed_transaction_rate_limiter->is_limited() ) {
+			throw new StateTransitionException(
+				__( 'Your payment was not processed.', 'woocommerce-payments' ),
+				400
+			);
```

Comment:
In addition to throwing the exception, which will reach the customer, please implement this code from the `catch()` block of the gateway here as well: https://github.com/Automattic/woocommerce-payments/blob/a19f2b0f38abdbce2946103fc9cfabe2d42684da/includes/class-wc-payment-gateway-wcpay.php#L1033-L1051

It's an action that we can perform immediately, and should not be a part of the catch block.

Reply:
Handled c0a8ddff5a3f02dfeb90950d99e90e6861a9c788 - note that I moved most of the logic to generate the note content to `OrderService` to simplify code in `InitialState`. Otherwise, it would be very verbose in `InitialState` and this state requires an additional dependency (`LegacyProxy`).